scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -400,6 +400,8 @@ def set_value(self, vid, value, subgroup=None, ignore_type=True): # pylint: disa
         for node in valnodes:
             self.set_text(node, value)
 
+        return value if valnodes else None
```

Comment:
This was a serious error. All of the subclasses of GenericXML return something if the set_value was successful, but this one wasn't doing that. Some of the subclasses were invoking this version and so this was a bug.